core/src/main/java/com/epam/coroutinecache/core/DiskCache.kt

```diff
@@ -66,44 +66,17 @@ class DiskCache(
         }
     }
 
-    override fun <T> getRecord(key: String): Record<T>? {
+    override fun <T : Any> getRecord(key: String, entryType: Type): Record<T>? {
         synchronized(this) {
-            try {
+            return try {
                 val safetyKey = safetyKey(key)
                 val resultedFile = File(cacheDirectory, safetyKey)
-                val type = jsonMapper.newParameterizedType(Record::class.java, Object::class.java)
-                val tempData: Record<T>? = jsonMapper.fromJson(resultedFile, type)
-
-                val dataClassName = if (tempData?.getDataClassName() == null) Object::class.java else Class.forName(tempData.getDataClassName())
-                val collectionClassName = if (tempData?.getDataCollectionClassName() == null) Object::class.java else Class.forName(tempData.getDataCollectionClassName())
-
-                val isCollection = Collection::class.java.isAssignableFrom(collectionClassName)
-                val isArray = collectionClassName.isArray
-                val isMap = Map::class.java.isAssignableFrom(collectionClassName)
-                val typeRecord: ParameterizedType
-
-                when {
-                    isCollection -> {
-                        val typeCollection = jsonMapper.newParameterizedType(collectionClassName, dataClassName)
-                        typeRecord = jsonMapper.newParameterizedType(Record::class.java, typeCollection)
-                    }
-                    isArray -> {
-                        typeRecord = jsonMapper.newParameterizedType(Record::class.java, collectionClassName)
-                    }
-                    isMap -> {
-                        val classKeyMap = Class.forName(tempData?.getDataKeyMapClassName())
-                        val typeMap = jsonMapper.newParameterizedType(collectionClassName, classKeyMap, dataClassName)
-                        typeRecord = jsonMapper.newParameterizedType(Record::class.java, typeMap)
-                    }
-                    else -> {
-                        typeRecord = jsonMapper.newParameterizedType(Record::class.java, dataClassName)
-                    }
-                }
-                val diskRecord: Record<T>? = jsonMapper.fromJson(resultedFile.absoluteFile, typeRecord)
+                val type = jsonMapper.newParameterizedType(Record::class.java, entryType)
```

Comment:
@vlivanov 
Currently here the only one place where newParameterizedType from jsonMapper interface used (just forgot to replace).
obtainTypeFromAnnotation method relies on methods from our Types object and I don't know any reason to keep this method in JsonMapper interface, so I suggest to remove this method from JsonMapper interface and replace with Type's object method call. Or rely only on interface's method and add one more method to JsonMapper interface - arrayOf that must return proper type for Array. What do you think?